src/commands/toolchain.rs

```diff
@@ -45,7 +45,7 @@ fn name_allowed(s: &str) -> Result<String> {
         None => s,
     };
 
-    if RESERVED_TOOLCHAIN_NAMES.contains(&name) {
+    if CHANNELS.contains(&name) {
```

Comment:
`RESERVED_TOOLCHAIN_NAMES` is different from `CHANNELS`. We want `stable` to be reserved even though it isn't currently a usable channel.